app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java

```diff
@@ -156,17 +200,84 @@ public void updateIfNeeded() {
             changed = true;
         }
 
+        /* Section - options */
         // Disable weak hash benchmark for faster startup.
         // https://github.com/syncthing/syncthing/issues/4348
         Element options = (Element) mConfig.getDocumentElement()
                 .getElementsByTagName("options").item(0);
         changed = setConfigElement(options, "weakHashSelectionMethod", "never") || changed;
 
+        /* Check if we have to dismiss any specific "unackedNotificationID" */
+        /* Dismiss "fsWatcherNotification" according to https://github.com/syncthing/syncthing-android/pull/1051 */
+        /* If "unackedNotificationID" would end up with an empty string or whitespace, remove the whole node */
+        if (getConfigElement(options, "unackedNotificationID").contains("fsWatcherNotification")) {
+            Log.i(TAG, "Remove 'fsWatcherNotification' from 'unackedNotificationID' node.");
+            String unackedNotificationID = getConfigElement(options, "unackedNotificationID").replace("fsWatcherNotification", "");
+            if (TextUtils.isEmpty(unackedNotificationID.replaceAll("\\s+",""))) {
+                options.removeChild(options.getElementsByTagName("unackedNotificationID").item(0));
+            } else {
+                setConfigElement(options, "unackedNotificationID", unackedNotificationID);
+            }
```

Comment:
I think there is still a misconception here. There is no need to do manipulate any strings: One unackedNotification node contains only one notification ID. So in the beginning it could look like this:
```xml
   <unackedNotificationID>otherNotification1</unackedNotificationID>
   <unackedNotificationID>fsWatcherNotification</unackedNotificationID>
   <unackedNotificationID>otherNotification2</unackedNotificationID>
   <unackedNotificationID>otherNotification3</unackedNotificationID>
```
and the result should be 
```xml
   <unackedNotificationID>otherNotification1</unackedNotificationID>
   <unackedNotificationID>otherNotification2</unackedNotificationID>
   <unackedNotificationID>otherNotification3</unackedNotificationID>
```
So you need to get the node that contains "fsWatcherNotification", which is not necessarily `.item(0)`, and remove the node, not the "fsWatcherNotification" string.

Reply:
I am on it. Incoming commit ...